Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py

```diff
@@ -1122,6 +1122,74 @@ def test_build_search_human_readable(mocker):
     assert headers == expected_headers
 
 
+def test_build_search_kwargs():
+    """
+    Given:
+        The command args.
+
+    When:
+        Running the build_search_kwargs to build the search query kwargs.
+
+    Then:
+        Ensure the query kwargs as expected.
+
+    """
+    args = {'earliest_time': '2021-11-23T10:10:10', 'latest_time': '2021-11-23T10:10:20', 'app': 'test_app', 'polling': False}
+    kwargs_normalsearch = splunk.build_search_kwargs(args)
+    for field in args:
+        if field == 'polling':
+            assert 'exec_mode' in kwargs_normalsearch
+        else:
+            assert field in kwargs_normalsearch
+
+
+@pytest.mark.parametrize('polling,status', [
+    (False, 'DONE'), (True, 'DONE'), (True, 'RUNNING')
+])
+def test_splunk_search_command(mocker, polling, status):
+    """
+    Given:
+        A search query with args.
+
+    When:
+        Running the splunk_search_command with and without polling.
+
+    Then:
+        Ensure the result as expected in polling and in regular search.
+
+    """
+    class Jobs:
+        def __init__(self):
+            self.oneshot = None
+
+        def create(query, earliest_time):
+            return {'sid': '123456', 'resultCount': 0}
+
+    class Service:
+        def __init__(self):
+            self.jobs = Jobs()
+
+        def job(sid, a):
+            return {'sid': sid, 'resultCount': 0}
+
+    mocker.patch.object(demisto, 'args', return_value={'query': 'query', 'earliest_time': '2021-11-23T10:10:10',
```

Comment:
note that our convention is to always pass demito.args() as part of the function arguments in order to avoid the mock in the UTs. no need to change, but keep in mind :)